components/idp-mgt/org.wso2.carbon.idp.mgt/src/main/java/org/wso2/carbon/idp/mgt/IdentityProviderManager.java

```diff
@@ -1007,6 +1015,215 @@ public void updateResidentIdP(IdentityProvider identityProvider, String tenantDo
+                    ((ExpressionNode) node).setValue(IdPManagementConstants.IS_FALSE_VALUE);
+                } else {
+                    if (log.isDebugEnabled()) {
+                        log.debug("Invalid \'isEnabled\' value passed in the filter. It should be \'true\' or " +
+                                "\'false\' isEnabled = " + ((ExpressionNode) node).getValue());
+                    }
+                    return;
+                }
+            }
+            expression.add((ExpressionNode) node);
+        } else if (node instanceof OperationNode) {
+            setExpressionNodeList(node.getLeftNode(), expression);
+            setExpressionNodeList(node.getRightNode(), expression);
+        }
+    }
+
+    /**
+     * Set the passing parameters as result.
+     *
+     * @param limit     page limit.
+     * @param offset    offset value.
+     * @param filter    filter value for IdP search.
+     * @param sortOrder order of IdP(ASC/DESC).
+     * @param sortBy    the column value need to sort.
+     * @param result    result object.
+     */
+    private void setParameters(int limit, int offset, String filter, String sortOrder, String sortBy, IdpSearchResult
+            result) {
+
+        result.setLimit(validateLimit(limit));
+        result.setOffSet(validateOffset(offset));
+        result.setSortBy(validateSortBy(sortBy));
+        result.setSortOrder(validateSortOrder(sortOrder));
+        result.setFilter(filter);
+    }
+
+    /**
+     * Validate sortBy.
+     *
+     * @param sortBy sortBy attribute.
+     * @return Validated sortOrder and sortBy.
+     */
+    private String validateSortBy(String sortBy) {
+
+        if (StringUtils.isBlank(sortBy)) {
+            if (log.isDebugEnabled()) {
+                log.debug("sortBy attribute is empty. Therefore we set the default sortBy attribute. sortBy" + sortBy);
+            }
+            return IdPManagementConstants.DEFAULT_SORT_BY;
+        } else {
+            switch (sortBy) {
+                case IdPManagementConstants.IDP_NAME:
+                    sortBy = IdPManagementConstants.NAME;
+                    break;
+                case IdPManagementConstants.IDP_HOME_REALM_ID:
+                    sortBy = IdPManagementConstants.HOME_REALM_ID;
+                    break;
+                default:
+                    sortBy = IdPManagementConstants.DEFAULT_SORT_BY;
+                    if (log.isDebugEnabled()) {
+                        log.debug("sortBy attribute is incorrect. Therefore we set the default sortBy attribute. " +
+                                "sortBy: " + sortBy);
+                    }
+                    break;
+            }
+            return sortBy;
+        }
+    }
+
+    /**
+     * Validate sortOrder.
+     *
+     * @param sortOrder sortOrder ASC/DESC.
+     * @return Validated sortOrder and sortBy.
+     */
+    private String validateSortOrder(String sortOrder) {
+
+        if (StringUtils.isBlank(sortOrder)) {
```

Comment:
Honor the accending value if comes in the request.

Reply:
This is done